### PR TITLE
update-check: T3476: Allow update-check for VyOS images

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

Ability to autocheck available new images
Parse remote URL JSON image-version.json file and compare version VyOS with a local current version, if find diff sent a wall
the message that the new image is available
Also, add the op-mode command to check images "show system updates"
With the option "auto-check," check will be once per 12 hours

```
  set system update-check auto-check
  set system update-check url 'http://example.com/image-version.json'
```
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T3476

## How to test
```
vyos@r14# set system update-check auto-check
[edit]
vyos@r14# set system update-check url 'http://192.168.122.14:8080/download/image-version.json'
[edit]
vyos@r14# commit
                                                                               
Update available: 1.4-rolling-202209060217 
Update URL: http://s3-us.vyos.io/rolling/current/vyos-rolling-latest.iso
                                                                               
[edit]
vyos@r14#
```
Op-mode:
```
vyos@r14:~$ show system updates 
Current version: 1.4-rolling-202209020217

Update available: 1.4-rolling-202209060217
Update URL: http://s3-us.vyos.io/rolling/current/vyos-rolling-latest.iso
vyos@r14:~$

```

Example of JSON remote file to parse/compare. It will be generated by Jenkins job in the future.
```
[
   {
      "arch":"amd64",
      "flavors":[
       "generic"
    ],
    "image":"vyos-rolling-latest.iso",
    "latest":true,
    "lts":false,
    "release_date":"2022-09-02",
    "release_train":"sagitta",
    "url":"http://s3-us.vyos.io/rolling/current/vyos-rolling-latest.iso",
    "version":"1.4-rolling-202209060217"
  }
]

```
MOTD if update if available:
motd

GraphQL API request: 
```
curl --raw 'https://localhost/graphql' \
  -H 'Content-Type: application/json' \
  -d '{"query":" {ShowUpdateSystem (data: {key: \"foo\"}) {success errors data {result}}}"}'

{"data":{"ShowUpdateSystem":{"success":true,"errors":null,"data":{"result":{"error":false,"update_available":true,"local_version":"1.4-rolling-202209020217","remote_version":"1.4-rolling-202209060217","url":"http://s3-us.vyos.io/rolling/current/vyos-rolling-latest.iso"}}}}}
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by jestabro** [comment]
@sever-sever This looks really good; one suggestion is to use the opmode error UnconfiguredSubsystem here:
https://github.com/sever-sever/vyos-1x/blob/T3476/src/op_mode/system.py#L76-L77
I'm trying to make use of the error hierarchy where appropriate; error messages added to the error will be reported to the API client in the message field.

**Comment by sever-sever** [comment]
> @sever-sever This looks really good; one suggestion is to use the opmode error UnconfiguredSubsystem here: https://github.com/sever-sever/vyos-1x/blob/T3476/src/op_mode/system.py#L76-L77 I'm trying to make use of the error hierarchy where appropriate; error messages added to the error will be reported to the API client in the message field.

Done!

**Comment by jestabro** [comment]
I did a simple test with mock image-version.json and all looks good; also tested error message.

**Review by jestabro — APPROVED** [review]

**Comment by c-po** [comment]
What about specifying a default URL tvat holds the JSON file on the VyOS repositories?

File: interface-definitions/include/url.xml.i [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  <properties>
+    <help>Remote URL</help>
+    <valueHelp>
+      <format>url</format>
+      <description>Remote URL</description>
+    </valueHelp>
+    <constraint>
+      <regex>^https?:\/\/?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*(\:[0-9]+)*(\/.*)?</regex>
+    </constraint>
+    <constraintErrorMessage>Incorrect URL format</constraintErrorMessage>
```
Comment (dmbaturin) [inline interface-definitions/include/url.xml.i:12]: If we require it to be HTTPS, maybe we should change the message to "Incorrect HTTPS URL". But I wonder if we should limit it to HTTPS.

Comment (sever-sever) [reply interface-definitions/include/url.xml.i:12]: It can be HTTP or HTTPS

**Pull request merged**